conf/supervisord-dev.conf

```diff
@@ -3,15 +3,15 @@ nodaemon = true
 silent = true
 
 [program:web]
-command = pserve --reload conf/development-app.ini
+command = gunicorn --paste conf/development.ini --config conf/gunicorn-dev.conf.py
```

Comment:
h was the only one of our apps that was running Gunicorn indirectly via `pserve`. I've changed it to run `gunicorn` directly like our other apps do and to use `gunicorn`'s `--paste` option so that the Pyramid settings get read.

This is unusual for a Pyramid app: the Pyramid docs tell you to use `pserve`. But we've always run `gunicorn` directly (in all apps but h) and I'm not aware of anything that we're losing by not using `pserve`. May as well remove an unnecessary layer?

I've gone for a `gunicorn --paste conf/*.ini --config conf/gunicorn*.conf.py` command format that gives us both a PasteDeploy config file and a Gunicorn config file. Even though we (rightly) tend to use envvars most often for config settings, I do think we want both config files. Examples in the Pyramid docs and many Pyramid extensions tell you to use the PasteDeploy config file, and it's where we put our logging config. And the Gunicorn config file is the only Gunicorn config source that supports _all_ Gunicorn config settings. There are separate config files for dev because it tends to differ a lot, but QA/staging/production share the same config file (so any differences need to go in envvars).

The cookiecutter will put these `gunicorn --paste conf/*.ini --config conf/gunicorn*.conf.py` commands in the `supervisord[-dev].conf` files and then apps will be able to customise the configs by overriding the `conf/*` files and/or by using environment variables (including the `GUNICORN_CMD_ARGS` envvar).

### Notes on Gunicorn configuration

To understand Gunicorn configuration you need to read the [Running Gunicorn](https://docs.gunicorn.org/en/stable/run.html), [Configuration Overview](https://docs.gunicorn.org/en/stable/configure.html) and [Settings](https://docs.gunicorn.org/en/stable/settings.html) pages in the Gunicorn docs and combine them in your head. Here's a short summary:

Gunicorn reads configuration from five locations and merges them. You can use all five of these at once, each successive location overriding earlier ones:

1. Certain specific environment variables for specific settings. For example `PORT` (for the port part of the `bind` setting), `WEB_CONCURRENCY` (for the `workers` setting), and a few others. We use some of these (e.g. `WEB_CONCURRENCY`) in our QA and production environments. Not all settings have environment variables, only a small handful do. The environment variables aren't all listed in one place in the Gunicorn docs, mentions of them are spread throughout the [Settings page](https://docs.gunicorn.org/en/stable/settings.html).

2. The PasteDeploy config file (if any), e.g. `development.ini` or `production.ini`. There are two variations of this:

   1. You can run a command like `pserve *.ini` which uses [Pyramid's pserve](https://docs.pylonsproject.org/projects/pyramid/en/latest/pscripts/pserve.html) to run Gunicorn indirectly. The `[server:main]` block in the `*.ini` PasteDeploy config file tells Pyramid to use Gunicorn (instead of [Waitress](https://docs.pylonsproject.org/projects/waitress/en/latest/), which is Pyramid's default):
   
      ```ini
      [server:main]
      use = egg:gunicorn#main
      bind = unix:/tmp/gunicorn-web.sock
      proc_name = web
      workers = 3
      ```

      In this case Gunicorn can read some Gunicorn config settings from the `[server:main]` block in the PasteDeploy config file (such as `proc_name` and `workers = 3` in the example above). Some Gunicorn settings (e.g. server hooks) can't go in a PasteDeploy config file.
      
      Some Gunicorn features such as hot-reloading don't work when running Gunicorn via pserve (you have to use pserve's reloading instead).
      
      See [Running the Project Application](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/project.html#running-the-project-application) in Pyramid's docs and [Paste Deployment](https://docs.gunicorn.org/en/stable/run.html#paste-deployment) in Gunicorn's docs and also [Paster Applications](https://docs.gunicorn.org/en/stable/configure.html#paster-applications) in the Gunicorn docs.

   2. You can run Gunicorn with the [`--paste` or `--paster` option](https://docs.gunicorn.org/en/stable/settings.html#paste):
   
      ```
      gunicorn --paste development.ini [GUNICORN_OPTIONS]
      ```

      In this case you're running Gunicorn directly and not using pserve. Gunicorn will read certain settings from the PasteDeploy config file such as the path to the WSGI application function and the logging config. But when run in this way Gunicorn will **not** read general Gunicorn settings from the PasteDeploy config file's `[server:main]` section, so you have to put your Gunicorn settings somewhere else. Gunicorn hot reloading will work.  

3. The Gunicorn config file (if any). By default this is `gunicorn.conf.py` in the current working directory but it can be changed with the `-c` / `--config` argument to the `gunicorn` command. A Gunicorn config file is the only source that gives you access to _all_ Gunicorn settings.

4. The `GUNICORN_CMD_ARGS` environment variable which can contain `gunicorn` command line arguments, e.g. `GUNICORN_CMD_ARGS="--bind=127.0.0.1 --workers=3"`.

5. Command line arguments passed directly to the `gunicorn` command itself. These override everything else. Not all Gunicorn settings have command line arguments.